Re-enable devcontainer CI jobs once @playwright/browser-chromium install no longer hangs

## Background

On 2026-05-21, the two CI jobs that run inside a devcontainer — `Devcontainer Runs Tests` and `DoenetML to PreTeXt Tests` (`.github/workflows/ci.yml`) — started hanging indefinitely inside `npm ci`. Last good run was 2026-05-21 06:43 UTC, completing `npm ci` in 59 s; subsequent runs hung for the full 6 h workflow timeout.

#1163 (now merged) walked through the investigation in detail. The root cause is in `@playwright/browser-chromium@1.57.0`'s install script: it downloads the full 164.7 MiB Chromium archive in ~1.5 s, then hangs on a post-download step (likely browser-launch validation) for as long as the job is allowed to run. Confirmed from the verbose log:

```
01:32:12.40  npm info run @playwright/browser-chromium@1.57.0 install
01:32:12.67  Downloading Chromium 143.0.7499.4 (playwright build v1200)...
01:32:14.05  |...| 100% of 164.7 MiB
                              ← 22 min of silence ←
01:54:15.88  ##[error]The operation was canceled.
```

`@playwright/browser-chromium` is pulled transitively by `@vscode/test-web` (a root devDep used only by the VS Code extension's local browser-dev scripts). The package version is lockfile-pinned and the install script content is byte-identical between the run that succeeded and the runs that hang, so something runtime/network on Playwright's or Azure's side changed — not something we can fix from this repo.

## Workaround currently in place

Both devcontainer jobs are skipped with `if: false` in `.github/workflows/ci.yml`. They produce no checks, so the CI workflow concludes `success` and `publish.yml` (gated on `workflow_run.conclusion == 'success'`) can trigger normally on main.

Tried and rejected on the way to this fix:

- `PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1` — broke `pretext-export.test.ts`, which launches Chromium via webdriverio in `beforeAll` and needs the binary to exist.
- `continue-on-error: true` — does not suppress timeout-induced *cancellations* (only `failure` conclusions), so the workflow still concluded `failure` and blocked publishing.

## What to do when reverting

1. Check that `@playwright/browser-chromium@1.57.0`'s install no longer hangs by running the devcontainer jobs on a temporary branch. Confirm by visiting the URL `https://cdn.playwright.dev/dbazure/download/playwright/builds/chromium/1200/chromium-linux.zip` succeeds quickly from inside the container, and that an install completes in <2 min.
2. Delete the `if: false` line (and the surrounding comment) on each of the two jobs:
   - `test-doenetml-to-pretext-devcontainer` (`Devcontainer Runs Tests`)
   - `test-doenetml-to-pretext-full-devcontainer` (`DoenetML to PreTeXt Tests`)
3. Optionally also drop `timeout-minutes: 25` (added as a safety bound) if the jobs are running cleanly again.

## Possible permanent fixes (so this doesn't recur)

If upstream takes a long time to resolve, or if we want to be insulated from similar future hangs in transitive postinstall scripts, options include:

- Point `webdriverio` in `packages/doenetml-to-pretext/test/utils/run-through-core.ts` at the Chromium already installed by the Python feature at `/opt/playwright-browsers/chromium-1223/chrome-linux/chrome` (via `goog:chromeOptions.binary`), and then skip the npm-side Playwright download with `PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1`.
- Move `@vscode/test-web` from the root `package.json` into `packages/vscode-extension`'s own `devDependencies` so it's not part of the workspace install at all.
- Replace the devcontainer-based jobs with regular ubuntu-runner jobs (the underlying tests are already covered by `Test Main` on the plain runner — the devcontainer wrapper exists only to validate the contributor setup, which can be checked less frequently).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/.devcontainer/devcontainer.json
+++ b/.devcontainer/devcontainer.json
@@ -4,7 +4,7 @@
     "image": "mcr.microsoft.com/devcontainers/base:ubuntu-24.04",
     "features": {
         "ghcr.io/devcontainers/features/node:1": {
-            "version": "24.15.0"
+            "version": "lts"
         },
         "ghcr.io/devcontainers/features/rust:1": {
             "version": "latest"

--- a/.github/workflows/changesets-version-pr.yml
+++ b/.github/workflows/changesets-version-pr.yml
@@ -25,8 +25,7 @@ jobs:
             - name: Use Node.js 24
               uses: actions/setup-node@v6
               with:
-                  # See ci.yml for the rationale behind pinning to 24.15.0.
-                  node-version: 24.15.0
+                  node-version: 24.x
                   cache: "npm"
 
             - name: Install dependencies

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -32,11 +32,7 @@ jobs:
 
         strategy:
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
                 # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
 
         steps:
@@ -79,11 +75,7 @@ jobs:
         needs: build
         strategy:
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
                 # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
         steps:
             - uses: actions/checkout@v6
@@ -120,11 +112,7 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
                 # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
                 test-group: [group1, group2, group3, group4]
         steps:
@@ -162,11 +150,7 @@ jobs:
         strategy:
             fail-fast: false
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
                 # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
                 test-group:
                     [
@@ -261,11 +245,7 @@ jobs:
         if: needs.changes.outputs.prefigure == 'true'
         strategy:
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
         steps:
             - uses: actions/checkout@v6
             - name: Use Node.js ${{ matrix.node-version }}
@@ -360,11 +340,7 @@ jobs:
 
         strategy:
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
                 # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
 
         steps:
@@ -416,11 +392,7 @@ jobs:
         needs: build-docs
         strategy:
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
         steps:
             - uses: actions/checkout@v6
             - name: Use Node.js ${{ matrix.node-version }}
@@ -461,11 +433,7 @@ jobs:
 
         strategy:
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
 
         steps:
             - name: Checkout sources
@@ -517,11 +485,7 @@ jobs:
 
         strategy:
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
 
         steps:
             - uses: actions/checkout@v6
@@ -577,11 +541,7 @@ jobs:
 
         strategy:
             matrix:
-                # Pinned to 24.15.0: Node 24.16.0 has an extract-zip regression
-                # (https://github.com/nodejs/node/issues/63487) that hangs
-                # cypress/playwright install scripts, freezing `npm ci` indefinitely.
-                # Revisit once a fixed 24.x release is available in setup-node.
-                node-version: [24.15.0]
+                node-version: [24.x]
 
         steps:
             - uses: actions/checkout@v6

--- a/.github/workflows/gh-pages-docs.yml
+++ b/.github/workflows/gh-pages-docs.yml
@@ -27,8 +27,7 @@ jobs:
 
         strategy:
             matrix:
-                # See ci.yml for the rationale behind pinning to 24.15.0.
-                node-version: [24.15.0]
+                node-version: [24.x]
                 # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
 
         steps:

--- a/.github/workflows/publish-prefigure.yml
+++ b/.github/workflows/publish-prefigure.yml
@@ -44,8 +44,7 @@ jobs:
             - name: Use Node.js 24
               uses: actions/setup-node@v6
               with:
-                  # See ci.yml for the rationale behind pinning to 24.15.0.
-                  node-version: 24.15.0
+                  node-version: 24.x
                   registry-url: https://registry.npmjs.org
 
             - name: Check whether current version is already published
@@ -159,8 +158,7 @@ jobs:
             - name: Use Node.js 24
               uses: actions/setup-node@v6
               with:
-                  # See ci.yml for the rationale behind pinning to 24.15.0.
-                  node-version: 24.15.0
+                  node-version: 24.x
                   registry-url: https://registry.npmjs.org
                   cache: "npm"
 

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -56,8 +56,7 @@ jobs:
             - name: Use Node.js 24
               uses: actions/setup-node@v6
               with:
-                  # See ci.yml for the rationale behind pinning to 24.15.0.
-                  node-version: 24.15.0
+                  node-version: 24.x
                   registry-url: https://registry.npmjs.org
                   cache: "npm"
 
@@ -138,7 +137,7 @@ jobs:
             - name: Use Node.js 24
               uses: actions/setup-node@v6
               with:
-                  node-version: 24.15.0
+                  node-version: 24.x
                   cache: "npm"
 
             - name: Install dependencies
@@ -228,8 +227,7 @@ jobs:
             - name: Use Node.js 24
               uses: actions/setup-node@v6
               with:
-                  # See ci.yml for the rationale behind pinning to 24.15.0.
-                  node-version: 24.15.0
+                  node-version: 24.x
                   registry-url: https://registry.npmjs.org
                   cache: "npm"
 

